### PR TITLE
Fix pollard serialization/deserialization (#46), Debug/Display for empty pollard (#48)

### LINKED ISSUE #46: Serialization/Deserialization fails for empty node hashes
```
use rustreexo::accumulator::{node_hash::NodeHash, pollard::Pollard};

#[test]
fn test_pollard_serialization() {
    let mut p = Pollard::new();
    let values = vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
    let hashes: Vec<NodeHash> = values
         .into_iter()
         .map(|i| NodeHash::from([i; 32]))
         .collect();
    p.modify(&hashes, &[]).expect("modify should work");
    p.modify(&[], &hashes).expect("modify should work");
    let mut serialized= Vec::<u8>::new();
    p.serialize(&mut serialized).expect("serialize should work");
    let _pollard = Pollard::deserialize(&*serialized).expect("deserialization fails");
}
```

### PR DESCRIPTION
**Pull request by Ash-L2L**
fixes #46, #48.
This fix is not backwards-compatible as the serialization encoding is changed. I have bumped the version number to reflect this.

### CODE REVIEW

**Comment by Davidson-Souza** [comment]
Thank you for this PR! Just some comments

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "rustreexo"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (Davidson-Souza) [inline Cargo.toml:3]: Not sure if we should do this here. We probably need another release, but let's do it in another pr

File: src/accumulator/pollard.rs [hunk 2]
```diff
@@ -622,6 +625,7 @@ mod test {
     use std::convert::TryFrom;
     use std::rc::Rc;
     use std::str::FromStr;
+    use std::task::Poll;
```
Comment (Davidson-Souza) [inline src/accumulator/pollard.rs:628]: I think this shouldn't be here.

File: src/accumulator/pollard.rs [hunk 3]
```diff
@@ -972,4 +976,25 @@ mod test {
...
+        let hashes: Vec<NodeHash> = values
+            .into_iter()
+            .map(|i| NodeHash::from([i; 32]))
+            .collect();
+        p.modify(&hashes, &[]).expect("modify should work");
+        p.modify(&[], &hashes).expect("modify should work");
+        let mut serialized = Vec::<u8>::new();
+        p.serialize(&mut serialized).expect("serialize should work");
+        let _deserialized = Pollard::deserialize(&*serialized).expect("deserialize should work");
+    }
```
Comment (Davidson-Souza) [inline src/accumulator/pollard.rs:999]: Can we `assert` the roots here?

**Review by Davidson-Souza — CHANGES_REQUESTED** [review]

File: src/accumulator/pollard.rs (hunk 3 above)
Comment (Ash-L2L) [reply src/accumulator/pollard.rs:1007]: Done, resolving

File: src/accumulator/pollard.rs (hunk 2 above)
Comment (Ash-L2L) [reply src/accumulator/pollard.rs]: Removed, resolving

**Review by Davidson-Souza — APPROVED** [review]

**Pull request merged**